site/src/main/java/uk/nhs/digital/ps/directives/TruncateFormatterDirective.java

```diff
@@ -0,0 +1,70 @@
+package uk.nhs.digital.ps.directives;
+
+import freemarker.core.Environment;
+import freemarker.template.*;
+
+import java.io.IOException;
+import java.util.Map;
+
+import static java.text.MessageFormat.format;
+
+/**
+ * Example usage in Freemarker template:
+ * <pre>
+ *
+ *     <#assign truncate="uk.nhs.digital.ps.directives.TruncateFormatterDirective"?new() >
+ *         ...
+ *     &lt;p&gt;<@truncate text=document.someText size=300/>.&lt;/p&gt;
+ * </pre>
+ *
+ */
+public class TruncateFormatterDirective implements TemplateDirectiveModel {
+
+    private static final String MAX_SIZE_PARAM_NAME = "size";
+    private static final String INPUT_PARAM_NAME = "text";
+
+    @Override
+    public void execute(final Environment environment, final Map parameters, final TemplateModel[] loopVariables,
+                        final TemplateDirectiveBody body) throws TemplateException, IOException {
+
+        assertRequiredParameterPresent(parameters, environment);
+
+        final int size = getValueAsInt(parameters, MAX_SIZE_PARAM_NAME);
+        final String text = getValueAsString(parameters, INPUT_PARAM_NAME);
+        final String truncatedSummary = truncate(text, size);
+
+        environment.getOut().append(truncatedSummary);
+    }
+
+    private String truncate(String text, final int size) {
+        int endIndex = Math.max(size, text.indexOf(" ", size));
+
+        if (endIndex < text.length()) {
+            return text.substring(0, endIndex) + "...";
+        } else {
+            return text;
+        }
+    }
+
+    private int getValueAsInt(final Map parameters, final String paramName) {
+        return Integer.parseInt(parameters.get(paramName).toString());
+    }
+
+    private String getValueAsString(final Map parameters, final String paramName) {
+        return parameters.get(paramName).toString();
+    }
+
+    private void assertRequiredParameterPresent(final Map parameters, final Environment environment)
+        throws TemplateException {
+
+        if (!parameters.containsKey(MAX_SIZE_PARAM_NAME)) {
```

Comment:
Suggest extracting this reusable block into a helper and then calling it directly from execute()